setImmediate.js

```diff
@@ -35,13 +32,13 @@
         if (currentlyRunningATask) {
             // Delay by doing a setTimeout. setImmediate was tried instead, but in Firefox 7 it generated a
             // "too much recursion" error.
-            setTimeout(partiallyApplied(runIfPresent, handle), 0);
+            setTimeout(function () { runIfPresent(handle); }, 0);
         } else {
             var task = tasksByHandle[handle];
             if (task) {
                 currentlyRunningATask = true;
                 try {
-                    task();
+                    run(task);
```

Comment:
ha fun fact from testing, try/finally is not an optimization killer...in v8, it is one in firefox